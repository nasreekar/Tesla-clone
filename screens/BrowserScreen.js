import React from 'react'
import { StyleSheet, View } from 'react-native';
import { WebView } from 'react-native-webview';

const BrowserScreen = ({route}) => {
    const { redirectUrl } = route.params;
    return(
        <View style = {styles.container}>
            <WebView 
                style={styles.webview}
                source={{ uri: redirectUrl }}
            />
       </View>
    )
}

export default BrowserScreen

const styles = StyleSheet.create({
    container: {
        top: '5%',
        height: '100%'
    },
    webview: {
        flex: 1
    }
 })