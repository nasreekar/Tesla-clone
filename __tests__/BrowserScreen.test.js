import React from 'react';
import { render } from "@testing-library/react-native";
import BrowserScreen from '../screens/BrowserScreen';

beforeEach(() => {
    jest.useFakeTimers();
  });
  

describe('<BrowserScreen />', () => {
    let snapshot;

    const mockedParams = {
        route: { params: { redirectUrl: 'www.google.com' } },
        navigation: ''
      };

    it('renders correctly', () => {
        snapshot = render(<BrowserScreen {...mockedParams} />);
        expect(snapshot.toJSON()).toMatchSnapshot();
      });

      it('should set webview uri', () => {
        snapshot = render(<BrowserScreen {...mockedParams} />);
        const child = snapshot.toJSON().children[0].children[0].props.source;
        expect(child.uri).toEqual('www.google.com');
      });
});