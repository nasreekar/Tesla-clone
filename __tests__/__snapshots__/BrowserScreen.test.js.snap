// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrowserScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "top": "5%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "uri": "www.google.com",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "flex": 1,
          },
        ]
      }
      useSharedProcessPool={true}
    />
  </View>
</View>
`;
